driver/main.c

```diff
@@ -2945,6 +2931,8 @@ MODULE_INFO(build_commit, DRIVER_COMMIT);
 MODULE_INFO(api_version, PPM_API_CURRENT_VERSION_STRING);
 MODULE_INFO(schema_version, PPM_SCHEMA_CURRENT_VERSION_STRING);
 
+module_param(per_cpu_buffer_dim, ulong, 0644);
```

Comment:
Neat!
However, what would happen if a second consumer (or just me, by manually `echo`ing a different number in `/sys/module/scap/parameters/per_cpu_buffer_dim`) changes the value? Would we read out of bounds? Would we cause a system crash?

Reply:
@FedeDP unfortunately yes, this approach is based on the idea that all the consumers have the same buffer dimension, that in my mind doesn't seem so strange, but if we want something different we have to change a lot of our kmod code, that' s never a good idea :thinking: If you have the privilege to change it manually your system is already gone probably ...

Reply:
What happens if I open multiple consumers? Either each of them can use a different ring size or they cannot. If 1) we need the changes anyway (AIUI it's basically ring->buffer_dim instead of a global). If 2) we must ignore any changes to the sysctl after the first consumer starts.

Reply:
The idea is (2) all the consumers have the same size, and yes now I understood that your worries were about the `sysctl` changes, in this case, yes we can consider this value only at open time as @FedeDP  as suggested

Reply:
How will the second consumer learn the ring size? They cannot rely on the sysctl.

TBH it seems much simpler to:
- copy the sysctl value to the ring/consumer in ppm_open
- use the copied value instead of the global

(as much as I'd prefer an ioctl I understand it's more work and now might not be the time for it)

Then every consumer has its own size of the rings and nobody really cares (AFAIK the rings aren't synchronized in any way and we redo all the filler work for every consumer)